pkg/config/config.go

```diff
@@ -0,0 +1,266 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"golang.org/x/oauth2"
+)
+
+const (
+	configFileName = "config.json"
+	configDirName  = "obsctl"
+)
+
+// getConfigFilePath returns the obsctl config file path or the first string in override argument.
+// Useful for testing.
+func getConfigFilePath(override ...string) string {
+	if len(override) != 0 {
+		return override[0]
+	}
+
+	usrConfigDir, err := os.UserConfigDir()
+	if err != nil {
+		return configFileName
+	}
+
+	return filepath.Join(usrConfigDir, configDirName, configFileName)
+}
+
+func ensureConfigDir() error {
+	if err := os.MkdirAll(path.Dir(getConfigFilePath()), 0700); err != nil {
+		return fmt.Errorf("creating config directory: %w", err)
+	}
+
+	return nil
+}
+
+type APIName string
+type TenantName string
```

Comment:
Just to have them separated in the `APIs` and `Contexts` maps. Otherwise, both would be map[string]Type. cc: @onprem 🙂

Reply:
Hm, is that an issue? Or do you mean this enhances readability?

Reply:
Enhances readability!

Reply:
Got it :+1: I see how it can help in some cases, but in other when we need to always modify the type like `config.APIName(<string>)` this is on the other hand adding extra cognitive load and more noise. So for me it kind of cancels out the benefits / downsides.

Reply:
Have removed them!